src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -102,7 +106,26 @@ public static DatasetTable tableModelToTable(TableModel tableModel) {
             .collect(Collectors.toList());
         datasetTable.primaryKey(primaryKeyColumns);
 
-        return datasetTable.columns(columns);
+        BigQueryPartitionConfig partitionConfig;
+        switch (tableModel.getPartitionMode()) {
+            case DATE:
+                String column = tableModel.getDatePartitionOptions().getColumn();
+                boolean useIngestDate = column.equals(PdaoConstant.PDAO_INGEST_DATE_COLUMN_ALIAS);
+                partitionConfig = useIngestDate ?
+                    BigQueryPartitionConfig.ingestDate() :
+                    BigQueryPartitionConfig.date(column);
+                break;
+            case INT:
+                IntPartitionOptionsModel options = tableModel.getIntPartitionOptions();
+                partitionConfig = BigQueryPartitionConfig.intRange(
+                    options.getColumn(), options.getMin(), options.getMax(), options.getInterval());
+                break;
+            default:
+                partitionConfig = BigQueryPartitionConfig.none();
+                break;
+        }
+
+        return datasetTable.bigQueryPartitionConfig(partitionConfig).columns(columns);
     }
 
     public static TableModel tableModelFromTable(DatasetTable datasetTable) {
```

Comment:
Note to self: Should update this method too, so partitioning settings flow back the other way?